### PR TITLE
Add -git version of GLFW

### PR DESCRIPTION
**Pull request by orowith2os**
Tested and working with Minecraft 1.20.

Needs the `-Dorg.lwjgl.glfw.libname=/app/lib/libglfw.so` java argument to work, as the "use system installation of GLFW" option is broken on newer Minecraft versions.

### CODE REVIEW

**Comment by orowith2os** [comment]
https://github.com/Admicos/minecraft-wayland/issues/45

**Comment by orowith2os** [comment]
bot, build org.prismlauncher.PrismLauncher

**Comment by orowith2os** [comment]
@Scrumplex ready for review & merge

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "shared-modules"]
 	path = shared-modules
 	url = https://github.com/flathub/shared-modules.git
+[submodule "minecraft-wayland"]
```
Comment (guihkx) [inline .gitmodules:4]: Looks like this submodule is not being used anywhere.

Comment (orowith2os) [reply .gitmodules]: The plan seems to initially be to use the patches from there, but I rebased em and put em here locally instead. Removed it.

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,4 @@
 [submodule "shared-modules"]
 	path = shared-modules
 	url = https://github.com/flathub/shared-modules.git
+
```
Comment (guihkx) [inline .gitmodules:4]: Since you're not including the submodule anymore, you can revert this particular change to this file.

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ build.sh
 .flatpak-builder
 flatrepo
 *.flatpak
+.buildconfig
```
Comment (guihkx) [inline .gitignore:6]: Nit: Does `flatpak-builder` or some other Flatpak tool create this file?

File: org.prismlauncher.PrismLauncher.yml [hunk 4]
```diff
@@ -58,8 +60,31 @@ modules:
       - mv /app/jre /app/jdk/17
       - /usr/lib/sdk/openjdk8/install.sh
       - mv /app/jre /app/jdk/8
+    cleanup: [/jre]
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:63]: Nit: This change reverts a styling change I introduced in 675be8f8f193c6e413168f92e51de60317ca61b7, so for consistency with other modules, I'd keep YAML arrays as lists, i.e.:

```suggestion
    cleanup:
      - /jre
```

File: org.prismlauncher.PrismLauncher.yml [hunk 5]
```diff
@@ -58,8 +60,31 @@ modules:
...
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DBUILD_SHARED_LIBS:BOOL=ON
+      - -DGLFW_USE_WAYLAND=ON
+    sources:
+      - type: git
+        url: https://github.com/glfw/glfw.git
+        commit: 3fa2360720eeba1964df3c0ecf4b5df8648a8e52
+      - type: patch
+        path: 0003-Don-t-crash-on-calls-to-focus-or-icon.patch
+
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:77]: Nit: Remove empty new line.

File: org.prismlauncher.PrismLauncher.yml [hunk 6]
```diff
@@ -58,8 +60,31 @@ modules:
...
+      - type: patch
+        path: 0005-Add-warning-about-being-an-unofficial-patch.patch
+
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:80]: Nit: Remove empty new line.

File: org.prismlauncher.PrismLauncher.yml [hunk 7]
```diff
@@ -58,8 +60,31 @@ modules:
...
+      - type: patch
+        path: 0007-Platform-Prefer-Wayland-over-X11.patch
+
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:83]: Nit: Remove empty new line.

File: org.prismlauncher.PrismLauncher.yml [hunk 8]
```diff
@@ -29,6 +29,8 @@ finish-args:
 modules:
   # Might be needed by some Controller mods (see https://github.com/isXander/Controlify/issues/31)
   - shared-modules/libusb/libusb.json
+  # Needed for proper Wayland support
+  - libdecor.json
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:33]: Nit: Separate modules by an empty line. :P

**Review by guihkx — COMMENTED** [review]
Just some very small things I noticed... Feel free to dismiss my nitpicks. :sweat_smile: 

Some other minor cosmetic changes I'd recommend:

- Moving patches into a separate `patches` folder
- Squashing all those commits into a single one before merging

File: org.prismlauncher.PrismLauncher.yml [hunk 9]
```diff
@@ -53,7 +56,6 @@ modules:
           url: https://api.github.com/repos/PrismLauncher/PrismLauncher/releases/latest
           version-query: .tag_name
           url-query: .assets[] | select(.name == "PrismLauncher-" + $version + ".tar.gz") | .browser_download_url
-
```
Comment (guihkx) [inline org.prismlauncher.PrismLauncher.yml:56]: Nit: Please undo this change.

**Review by guihkx — APPROVED** [review]
Minor nitpick aside, LGTM.

Though I'm unable to actually test this on a Wayland session, since I have a really old NVIDIA GPU.

**Comment by orowith2os** [comment]
Fixed the last nit, and I can confirm that it works. No issues noticed so far, only that it's a pain to use system GLFW - but that's an issue with all newer Minecraft versions, and not something specific to this hack.

**Review by Scrumplex — APPROVED** [review]
LGTM. Thanks!
